### PR TITLE
Fix Python error name regression

### PR DESCRIPTION
**Pull request by bendk**
I hit this one when trying to upgrade application-services to 0.28.1.  It's kind of surprising that this is the first we're hearing of it, but I guess the Python bindings users are a minority.

### CODE REVIEW

File: uniffi_bindgen/src/interface/visit_mut.rs [hunk 1]
```diff
@@ -131,6 +131,14 @@ impl ComponentInterface {
...
             }
         }
+
+        self.errors = self.errors
+            .drain()
+            .map(|mut name| {
+                visitor.visit_error_name(&mut name);
+                name
+            })
+            .collect()
```
Comment (bendk) [inline uniffi_bindgen/src/interface/visit_mut.rs:141]: This is the basic issue, this is a somewhat hacky field that stores error type names when we see them in "[Throws=]" UDL attributes.  I think that proc-macros might have handled this correctly, but I'm not sure.

File: uniffi_bindgen/src/bindings/python/gen_python/error.rs [hunk 2]
```diff
@@ -22,7 +22,7 @@ impl CodeType for ErrorCodeType {
     }
 
     fn canonical_name(&self) -> String {
-        format!("Type{}", self.id)
+        format!("Type{}", self.type_label())
```
[the remark this answers is not in the document]
Comment (bendk) [reply uniffi_bindgen/src/bindings/python/gen_python/error.rs:25]: Maybe I only needed to modify `ErrorCodeType::canonical_name`, but it seemed safer to change all of the class-like types to use the converted name.

**Review by gruberb — APPROVED** [review]
👍 LGTM

File: uniffi_bindgen/src/interface/visit_mut.rs (hunk 1 above)
Comment (gruberb) [reply uniffi_bindgen/src/interface/visit_mut.rs:142]: I like `drain` in combination with `collect`. I think I could have used something similar here as well!

**Pull request merged**